File: autoswap.js
require('dotenv').config();
const Web3 = require('web3');

// Load environment variables
const RPC_URL = process.env.RPC_URL;
const WALLET_ADDRESS = process.env.WALLET_ADDRESS;
const PRIVATE_KEY = process.env.PRIVATE_KEY;
const TOKEN_IN = process.env.TOKEN_IN;
const TOKEN_OUT = process.env.TOKEN_OUT;
const ROUTER_ADDRESS = process.env.ROUTER_ADDRESS;

const web3 = new Web3(RPC_URL);

// Router ABI (you need to replace this with the actual ABI of the router contract)
const routerAbi = [ /* Add the router contract ABI here */ ];
const routerContract = new web3.eth.Contract(routerAbi, ROUTER_ADDRESS);

// Swap function
async function autoSwap(amountIn, amountOutMin, deadline) {
    try {
        const tx = routerContract.methods.swapExactTokensForTokens(
            web3.utils.toWei(amountIn.toString(), 'ether'),
            web3.utils.toWei(amountOutMin.toString(), 'ether'),
            [TOKEN_IN, TOKEN_OUT],
            WALLET_ADDRESS,
            deadline
        );

        const gas = await tx.estimateGas({ from: WALLET_ADDRESS });
        const gasPrice = await web3.eth.getGasPrice();

        const txData = {
            from: WALLET_ADDRESS,
            to: ROUTER_ADDRESS,
            data: tx.encodeABI(),
            gas,
            gasPrice
        };

        const signedTx = await web3.eth.accounts.signTransaction(txData, PRIVATE_KEY);
        const receipt = await web3.eth.sendSignedTransaction(signedTx.rawTransaction);

        console.log('Swap successful:', receipt);
    } catch (error) {
        console.error('Error during swap:', error);
    }
}

// Example usage
const amountIn = 1; // Amount of TOKEN_IN to swap
const amountOutMin = 0.9; // Minimum amount of TOKEN_OUT to receive
const deadline = Math.floor(Date.now() / 1000) + 60 * 10; // 10 minutes from now

autoSwap(amountIn, amountOutMin, deadline);

File: autoswap.py
import os
from web3 import Web3
from dotenv import load_dotenv

# Load environment variables from .env
load_dotenv()

# Environment variables
RPC_URL = os.getenv("RPC_URL")
WALLET_ADDRESS = os.getenv("WALLET_ADDRESS")
PRIVATE_KEY = os.getenv("PRIVATE_KEY")
TOKEN_IN = os.getenv("TOKEN_IN")
TOKEN_OUT = os.getenv("TOKEN_OUT")
ROUTER_ADDRESS = os.getenv("ROUTER_ADDRESS")

# Web3 setup
web3 = Web3(Web3.HTTPProvider(RPC_URL))
if not web3.isConnected():
    raise ConnectionError("Failed to connect to blockchain")

# Router contract ABI (replace with actual ABI)
ROUTER_ABI = [
    # Add the ABI here
]

router_contract = web3.eth.contract(address=ROUTER_ADDRESS, abi=ROUTER_ABI)

def auto_swap(amount_in, amount_out_min, deadline):
    """
    Perform a token swap.
    """
    try:
        # Prepare transaction
        transaction = router_contract.functions.swapExactTokensForTokens(
            web3.toWei(amount_in, 'ether'),  # Amount of TOKEN_IN
            web3.toWei(amount_out_min, 'ether'),  # Minimum amount of TOKEN_OUT
            [TOKEN_IN, TOKEN_OUT],  # Path: TOKEN_IN -> TOKEN_OUT
            WALLET_ADDRESS,  # Recipient
            deadline  # Deadline (timestamp)
        )

        # Estimate gas and get gas price
        gas = transaction.estimateGas({'from': WALLET_ADDRESS})
        gas_price = web3.eth.gas_price

        # Build transaction
        tx = transaction.buildTransaction({
            'from': WALLET_ADDRESS,
            'gas': gas,
            'gasPrice': gas_price,
            'nonce': web3.eth.getTransactionCount(WALLET_ADDRESS)
